internal/uvm/scsi/mount.go

```diff
@@ -0,0 +1,169 @@
+package scsi
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"sync"
+
+	"github.com/Microsoft/hcsshim/internal/protocol/guestresource"
+)
+
+type mountManager struct {
+	m       sync.Mutex
+	mounter Mounter
+	// Tracks current mounts. Entries will be nil if the mount was unmounted, meaning the index is
+	// available for use.
+	mounts   []*mount
+	mountFmt string
+}
+
+func newMountManager(mounter Mounter, mountFmt string) *mountManager {
+	return &mountManager{
+		mounter:  mounter,
+		mountFmt: mountFmt,
+	}
+}
+
+type mount struct {
+	path       string
+	index      int
+	controller uint
+	lun        uint
+	config     *MountConfig
+	waitErr    error
+	waitCh     chan struct{}
+	refCount   uint
+}
+
+// MountConfig specifies the options to apply for mounting a SCSI device in
+// the guest OS.
+type MountConfig struct {
+	ReadOnly  bool
+	Encrypted bool
+	Verity    *guestresource.DeviceVerityInfo
+	Options   []string
+}
+
+func mountConfigEquals(a, b *MountConfig) bool {
+	return a.ReadOnly == b.ReadOnly &&
```

Comment:
We should also have this check in here right? 

```
if a == nil || b == nil {
		return a == nil && b == nil
	}
```

Reply:
Changed to use `reflect.DeepEqual`.